src/frontend/serialization_phase.nit

```diff
@@ -34,27 +34,118 @@ redef class ToolContext
 	private fun place_holder_type_name: String do return "PlaceHolderTypeWhichShouldNotExist"
 end
 
+redef class ANode
```

Comment:
There is a problem with AModuleDecl, that is not in the tree above each node, so I also redef AModule to redirect `is_serialize` calls to its `n_moduledecl`. So I will keep the introduction on `ANode` but will move the check on the annotation to ADefinition.